### ISSUE TITLE
Vue files are not type checked or linted with useTypescriptIncrementalApi

### ISSUE THREAD

**Issue by WolfspiritM** [TypeStrong/fork-ts-checker-webpack-plugin#219]
Hi,

My config is:

```
            new ForkTsCheckerWebpackPlugin({
                tsconfig: './tsconfig.json',
                tslint: './tslint.json',
                vue: true,
                async: false,
                formatter: "codeframe",
                useTypescriptIncrementalApi: true,
                measureCompilationTime: true
            }),
```

Using that config an error I implemented in a vue file is not checked:
e.g.: 

> const test: string = 1234562123;

Does not produce any error.

Changing useTypescriptIncrementalApi to false works but is much slower.

**Comment by johnnyreilly** [comment]
Thanks for reporting. We have integration tests that check that Vue errors etc are caught:

https://github.com/Realytics/fork-ts-checker-webpack-plugin/blob/master/test/integration/incrementalApi.spec.js

Could you compare your setup with this please? I'm keen we understand why you might not be picking up errors with your config. It might help you're able to produce a minimal reproduction of the issue and then compare that to our integration tests.

Neither myself or the implementors of this support are Vue users and so we may benefit from assistance with this. 😉

Any help you can provide would be greatly appreciated!

**Comment by WolfspiritM** [comment]
Thanks for the answer.

I can't really find a big difference when it comes to configuration but it seems there is no test that checks for type and linting errors directly in *.vue files (or I missed that).
Maybe it's also just related to the output/cli?

I made a sample project here:
https://github.com/WolfspiritM/simplevue

```
npm install
npm run build
```

In that code results with:

```
Starting type checking and linting service...
Using 1 worker with 2048MB memory limit
Hash: d042033d16e0ac6c5f8f
Version: webpack 4.29.5
Time: 2274ms
Built at: 2019-02-27 22:06:50
Entrypoint bundle = js/bundle.js
[0] multi ./scripts/app.ts 28 bytes {bundle} [built]
[./node_modules/webpack/buildin/global.js] (webpack)/buildin/global.js 472 bytes {bundle} [built]
[./scripts/app.ts] 170 bytes {bundle} [built]
    + 14 hidden modules
```

Even so the ./scripts/components/HelloWorld.vue file contains a type error.
Changing the webpack config to not use the new api and running build again the error is correctly reported.

**Comment by johnnyreilly** [comment]
Thanks so much for that. I think you may have identified a bug. If you (or anyone else) is able to take a look I'd greatly appreciate the assistance.

In the meantime I'll change the plugin so it doesn't default to use the incremental API with Vue projects. I'll make we get that in for the 1.0.0 release.

**Comment by 0xorial** [comment]
I had a fresh look at the tests, and vue tests for useTypescriptIncrementalApi are quite poor - there is [only one](https://github.com/Realytics/fork-ts-checker-webpack-plugin/blob/master/test/integration/incrementalApi.spec.js#L113) test actually testing for errors and it is wrong (there should 2 errors, not 1). 
@johnnyreilly, what do you think about applying same technique as in `index.spec.js` ? (creating two sets of tests for incremental compilation true/false )

**Comment by johnnyreilly** [comment]
Completely agree @0xorial. 

I believe @wolfspiritm has been working on trying to fix the issue here: https://github.com/Realytics/fork-ts-checker-webpack-plugin/pull/220

I'm not too sure how things are going.

**Comment by piotr-oles** [comment]
@WolfspiritM
Please try [fork-ts-checker-webpack-plugin@alpha](https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/tree/alpha) - I've published a new version which supports Vue.js with the Watch API  🚀(note: TypeScript >= 3.8.0 required)
I will close this issue to clean-up the backlog :)
